code/modules/shuttle/emergency.dm

```diff
@@ -226,24 +235,30 @@
 				mode = SHUTTLE_IDLE
 				timer = 0
 		if(SHUTTLE_CALL)
+			if(time_left <= 135 && crash && !crash_announced)
+				//Notify people that a coder is an absolute ass
+				event_announcement.Announce("An error has been detected in shuttle docking systems - brace for impact!", "Don't drink and drive", new_sound = 'sound/misc/notice1.ogg')
+				crash_announced = TRUE
 			if(time_left <= 0)
 				//move emergency shuttle to station
 				if(dock(shuttle_master.getDock("emergency_home")))
 					setTimer(20)
 					return
+				//Be a dick to everyone and ruin their day (okay, not everyone, only those who are not buckled)
+				if(crash)
+					var/area/target = locate(crash_target)
+					for(var/mob/M in target)
+						M << 'sound/effects/meteorimpact.ogg'
+						if(M.buckled || istype(M.loc, /turf/space))
```

Comment:
Wondering if this should also pass over mobs that are flying, laying down (HIT THE DECK!), and only affect people if the area has gravity at the time of impact

That might be getting a bit more involved, but you can check these with `if(M.buckled || M.lying || M.flying || !M.mob_has_gravity())`

Space turfs don't have gravity, so the gravity check will handle checking for that them, and active mag-boots will also save you from getting floored by the impact in this case due to them forcing the mob to have gravity even if the area doesn't.

Reply:
Hmm...